app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/actions/insertScriptValidators.js

```diff
@@ -0,0 +1,314 @@
+import { parse } from "acorn";
+import { simple } from "acorn-walk";
+import { HtmlValidate, StaticConfigLoader } from "html-validate";
+import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
+
+/* LOGICAL VALIDATORS - currently not being fully applied */
+export const scriptLogicalErrors = {
+  DOM_LOAD_EVENT_LISTENER_AFTER_PAGE_LOAD: "dom_load_event_listener_after_page_load",
+  CONTAINS_HTML_CODE: "contains_html_code",
+};
+
+export function checkForLogicalErrorsInCode(code, script) {
+  if (doesStringContainHTMLSnippet(code)) {
```

Comment:
`checkForNonJsOrCssCode` because that's the intent.

Reply:
renamed to `isJSOrCSS` instead